File: examples/simple.py
from pathlib import Path
from pyledown import PileParams, Strand, LibFragmentType
import pandas as pd
import pyarrow as pa
import seaborn as sns
import matplotlib.pyplot as plt

input_bam = Path() / "tests" / "data" / "SRR21778056-sorted-subsample.bam"

regions_df = pd.DataFrame({
    "seq": ["chr1", "chr1"],
    "start": [14000, 20000],
    "end": [20000, 25000],
    "name": ["region_a", "region_b"],
    "strand": ["-", "-"],
})

pld = PileParams(
    input_bam,
    LibFragmentType.Isr,
    regions_df=regions_df,
)

res = pld.generate()
res = res.set_column(4, "down", pa.compute.multiply(res["down"].cast("int64"), -1))

plot_region = res.filter(pa.compute.equal(res["name"], "region_a")).to_pandas()

sns.lineplot(plot_region, x="pos", y="up")
sns.lineplot(plot_region, x="pos", y="down")
plt.title("region_a: chr1:14000-20000 (reverse strand)")
plt.show()

File: examples/simple.R
library(piledownR)
library(arrow)
library(ggplot2)
library(tidyr)
library(dplyr)

input_bam <- file.path("tests", "data", "SRR21778056-sorted-subsample.bam")

# --- Coverage ---

regions <- tibble::tibble(
  seq = c("chr1", "chr1"),
  start = c(14000, 20000),
  end = c(20000, 25000),
  name = c("region_a", "region_b"),
  strand = c("-", "-")
)

params <- pile_params(
  input_bam = input_bam,
  lib_fragment_type = "isr",
  regions_df = regions
)

reader <- generate(params)
df <- as_tibble(reader$read_table())
df$down <- -df$down

plot_df <- df[df$name == "region_a", ]
long <- pivot_longer(plot_df, cols = c("up", "down"), names_to = "direction", values_to = "count")

ggplot(long, aes(x = pos, y = count, colour = direction)) +
  geom_line() +
  labs(x = "Position", y = "Coverage", title = "region_a: chr1:14000-20000 (reverse strand)") +
  theme_minimal()

# --- Junction counting ---

jp <- junction_params(
  input_bam = input_bam,
  lib_fragment_type = "isr",
  seqs = c("chr1", "chr1"),
  starts = c(153990803, 23694792),
  ends = c(153991114, 23695797),
  region_names = c("junc_a", "junc_b"),
  region_strands = c("+", "-")
)

junc_reader <- generate_stream(jp)
junc_df <- as_tibble(junc_reader$read_table())
print(junc_df)
